src/Http/Http.Results/src/ValidationProblem.cs

```diff
@@ -49,21 +49,25 @@ internal ValidationProblem(HttpValidationProblemDetails problemDetails)
     int? IStatusCodeHttpResult.StatusCode => StatusCode;
 
     /// <inheritdoc/>
-    public Task ExecuteAsync(HttpContext httpContext)
+    public async Task ExecuteAsync(HttpContext httpContext)
     {
         ArgumentNullException.ThrowIfNull(httpContext);
 
         var loggerFactory = httpContext.RequestServices.GetRequiredService<ILoggerFactory>();
         var logger = loggerFactory.CreateLogger(typeof(ValidationProblem));
+        var problemDetailsService = httpContext.RequestServices.GetService<IProblemDetailsService>();
 
         HttpResultsHelper.Log.WritingResultAsStatusCode(logger, StatusCode);
         httpContext.Response.StatusCode = StatusCode;
 
-        return HttpResultsHelper.WriteResultAsJsonAsync(
+        if (problemDetailsService is null || !await problemDetailsService.TryWriteAsync(new() { HttpContext = httpContext, ProblemDetails = ProblemDetails }))
```

Comment:
The performance regression here is a little unfortunate:
- async state machine
- ProblemsDetailsContext object

Is there no way to optimize this?

Reply:
@davidfowl I don't see any synchronous APIs on `IProblemDetailService`.

I imagine http results are very hot code, but this should be scoped to a particular class of failure results, shouldn't it?